src/main/java/org/webshar/hrms/service/LeaveApplicationService.java

```diff
@@ -124,7 +120,7 @@ public EmployeeLeaveApplicationResponse applyLeave(
           ErrorMessageConstants.LEAVE_INSUFFICIENT_LEAVE_COUNT_OF_GIVEN_LEAVE_TYPE);
     } else {
       isStartAndEndDateAreOverlappingWithExistingRecordForGivenEmployeeAndLeaveTypeId(
-          employee.getId(), leaveType.getId(), employeeLeaveApplicationCreateRequest.getStartDate(),
```

Comment:
Why employee.id and leaveType.id is removed here. This verification should also include those parameters.